R/zoop_downloader.R

```diff
@@ -183,10 +195,135 @@ Zoopdownloader <- function(
       dplyr::group_by(dplyr::across(-.data$CPUE))%>%
       dplyr::summarise(CPUE=sum(.data$CPUE, na.rm=TRUE))%>% #Some taxa now have the same names (e.g., CYCJUV and OTHCYCJUV) so we now add those categories together.
       dplyr::ungroup()%>%
+      dplyr::left_join(stations, by=c("Source", "Station"))%>% #Add lat and long
       tibble::as_tibble() #required to finish operation after lazy_dt()
 
     cat("\nEMP_Meso finished!\n\n")
   }
+
+
+  # DOP Meso ---------------------------------------------------------------------
+  if("DOP_Meso"%in%Data_sets) {
+
+    DOP_Meso_file<-"DOP_ICF_Mesozooplankton_Abundance2017-2021"
+    DOP_Meso_URL<-paste0(DOP_pkg_url, "/", DOP_entities[DOP_Meso_file])
+
+    DOP_trawls_file<-"DOP_ICF_TowData2017-2021"
+    DOP_trawls_URL<-paste0(DOP_pkg_url, "/", DOP_entities[DOP_trawls_file])
+
+    #download the files
+    if (!file.exists(file.path(Data_folder, DOP_Meso_file)) | Redownload_data) {
+      Tryer(n=3, fun=utils::download.file, url=DOP_Meso_URL,
+            destfile=file.path(Data_folder, DOP_Meso_file), mode="wb", method=Download_method)
+    }
+    if (!file.exists(file.path(Data_folder, DOP_trawls_file)) | Redownload_data) {
+      Tryer(n=3, fun=utils::download.file, url=DOP_trawls_URL,
+            destfile=file.path(Data_folder, DOP_trawls_file), mode="wb", method=Download_method)
+    }
+
+
+    # Import the DOP data
+
+    zoo_DOP_Meso<-readr::read_csv(file.path(Data_folder, DOP_Meso_file))
+    zoo_DOP_trawls<-readr::read_csv(file.path(Data_folder, DOP_trawls_file))
+
+    # Tranform from "wide" to "long" format, add some variables,
+    # alter data to match other datasets
+
+    data.list[["DOP_Meso"]] <- zoo_DOP_Meso %>%
+      tidyr::pivot_longer(cols = !ICF_ID, names_to = "DOP_Meso", values_to = "CPUE") %>%
+      dplyr::left_join(zoo_DOP_trawls) %>%
+      dplyr::mutate(Datetime =  lubridate::ymd(as.character(Date), tz = "America/Los_Angeles") + lubridate::hms(as.character(Start_Time)),
+              Source = "DOP", #add variable for data source
+              SizeClass = "Meso") %>%
+      dplyr::filter(!is.na(.data$Mesozooplankton_Volume)) %>% #get rid of environmental variables with no data
+
+      #Select variables we are interested in. I need to check on the latitude/longitude issue with Sam.
```

Comment:
We settled the Lat/Long issue right, or is there anything else you need to check on?

Reply:
Nope! We should be good, I'll delete that comment.